util/design/gen-rng-health-thresholds.py

```diff
@@ -0,0 +1,132 @@
+
+`--per-bit` must be set to reflect the `RNG_BIT_ENABLE` entropy src configuration.
+
+"""
+import argparse
+import math
+import sys
+
+# Matches the `RNG_BUS_WIDTH` in entropy_src_pkg.sv
+_RNG_BUS_WIDTH = 4
+
+# List of test windows supported by this script.
+_TEST_NAMES = ("ADAPTP", "BUCKET", "MARKOV")
+
+
+def threshold_calc(test: str, window_size: int, sigma: float,
+                   per_bit: bool) -> bool:
+    """Calculates and prints high and low entropy health test thresholds.
+
+    Args:
+        test: Test name.
+        window_size: Window size in bits.
+        sigma: Number of standard deviations to provide between the range. This
+        assumes that the window is large enough to treat the test as normally
+        distributed.
+        per_bit: Set to true to calculate thresholds on a per RNG bit basis.
+    Returns:
+        False if unable to calculate the thresholds. True otherwise.
+    """
+    n: int = 0
+    p: float = 0.5
+
+    if test == "ADAPTP":
+        n = (window_size / _RNG_BUS_WIDTH) if per_bit else window_size
+    elif test == "BUCKET":
+        n = (window_size / _RNG_BUS_WIDTH)
+        p = 1.0 / float(1 << _RNG_BUS_WIDTH)
+    elif test == "MARKOV":
+        half_window = window_size / 2
+        n = (half_window / _RNG_BUS_WIDTH) if per_bit else half_window
+    else:
+        print(f"Invalid test name {test}")
+        return False
+
+    mean = p * n
+    stddev = math.sqrt(p * (1 - p) * n)
+
+    low = 0 if test == "BUCKET" else math.floor(mean - sigma * stddev)
+    high = math.ceil(mean + sigma * stddev)
+
+    # For large values of sigman, the gaussian approximation can recommend
+    # thresholds larger than the total number of trials. In such cases, we
+    # cap the threshold at the total number of trials for the given test.
+    low = 0 if low < 0 else low
+    high = n if high > n else high
+
+    print(f"Results for window size: {window_size:d}, per_bit: {per_bit}")
+    print(f"{test}_HI: 0x{int(high):08x}")
+    print(f"{test}_LO: 0x{int(low):08x}")
+
+    return True
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        prog="gen-rng-health-thresholds",
+        description=__doc__,
+        formatter_class=argparse.RawDescriptionHelpFormatter)
+    parser.add_argument('--window_size',
+                        '-w',
+                        type=int,
+                        help='Window size in bits.')
+    parser.add_argument('--sigma',
+                        '-s',
+                        type=float,
+                        help='''
+                        Number of standard deviations to support in the test
+                        window threshold.
+                        ''')
+    parser.add_argument('--per_bit',
+                        '-b',
+                        action='store_true',
+                        help='''
+                        Set to true to make calculations assuming single bit
+                        entropy.
+                        ''')
```

Comment:
Should we have a defined `default` for these arguments? Or should we make the window size and sigma mandatory arguments?

Reply:
Done. Added defaults based on FIPS configuration.